src/main/java/picard/sam/markduplicates/UmiGraph.java

```diff
@@ -173,13 +193,32 @@ public UmiGraph(DuplicateSet set, String umiTag, String assignedUmiTag, boolean
             // then choose the one with the fewest Ns
             if (assignedUmi == null) { assignedUmi = fewestNUmi; }
 
-            // Set the records to contain the assigned UMI
+            // Set the records to contain the inferred UMI and the Unique Molecular Identifier
             for (final SAMRecord rec : recordList) {
                 if (allowMissingUmis && rec.getStringAttribute(umiTag).isEmpty()) {
                     // The SAM spec doesn't support empty tags, so we set it to null if it is empty.
                     rec.setAttribute(umiTag, null);
+
                 } else {
-                    rec.setAttribute(assignedUmiTag, assignedUmi);
+                    final String fragmentStartPosition;
+                    if (ds.getRepresentative().getReadNegativeStrandFlag()) {
+                        fragmentStartPosition = ds.getRepresentative().getContig() + ":" + ds.getRepresentative().getAlignmentStart();
```

Comment:
Are there constants that can be used for the magic strings `":"` and `"/"`? E.g. if any other code depends on these particular string values, it would be better to use a named constant in both places to make the relationship between the different pieces of code clearer.